### DISCUSSION TITLE
Will PhotoPrism run (smoothly) on a Raspberry Pi 3?

### DISCUSSION THREAD

**Discussion by kev153** [photoprism/photoprism#3110] [General]
Hi, I'm want to get PhotoPrism, but find it very hard to find a Raspberry Pi 4. They seem to be sold out everywhere and I also can't find any used ones for a decent price. So I'm wondering if it will also run on a Raspberry Pi 3 with only 1 GB of RAM. The[ official system requirements](https://docs.photoprism.app/getting-started/#system-requirements) recommend 3 GB, so 1 GB really seems too little to me. On the other hand, there are posts here and there about PhotoPrism on Raspberry Pi 3. Does anybody have experience with that and can give an overview of the limitations if any? Or tips how to improve performance on such a system?

**Comment by lastzero** [comment]
As always, this ultimately depends on your expectations and the number and type of files you have. So it's hard to give a satisfying answer. If you can avoid it, I would generally not use a Pi 3, as you will probably run into limitations and it won't do well with many files.

**Reply by kev153** [reply]
Thanks for your reply. I currently have about 150 GB of photos, I guess 90 % jpeg and 10 % raw. Would you consider that "many files"?

**Reply by lastzero** [reply]
In between I would say. I've never tested a Pi 3, but my feeling is that it shouldn't be more than ~50 GB.

**Reply by lastzero** [reply]
You can compensate lacking CPU power and memory with a fast SSD to some extent, but the USB ports are probably not that fast either.

**Reply by kev153** [reply]
The Pi 3 has USB 2.0 ports, which have a rate of 480 MBit/s, meaning that a fast SSD wouldn't be able to fulfil its full potential anyway.

**Reply by lastzero** [reply]
That would be the biggest problem then in terms of performance.
